docs/programming_guide/controllers/model_controller.rst

```diff
@@ -0,0 +1,236 @@
+              clients = self.sample_clients(self.num_clients)
+
+              # send model to target clients with default train task, wait to receive results
+              results = self.send_model_and_wait(targets=clients, data=model)
+
+              # use BaseFedAvg aggregate function
+              aggregate_results = self.aggregate(
+                  results, aggregate_fn=self.aggregate_fn
+              )  # using default aggregate_fn with `WeightedAggregationHelper`. Can overwrite self.aggregate_fn with signature Callable[List[FLModel], FLModel]
+
+              # update global model with agggregation results
+              model = self.update_model(model, aggregate_results)
+
+              # save model (by default uses persistor, can provide custom method)
+              self.save_model(model)
+
+          self.info("Finished FedAvg.")
+
+
+Below is a comprehensive table overview of the :class:`ModelController<nvflare.app_common.workflows.model_controller.ModelController>` API:
+
+
+.. list-table:: ModelController API
+   :widths: 25 35 50
+   :header-rows: 1
+
+   * - API
+     - Description
+     - API Doc Link
+   * - run
+     - Run routine for workflow.
+     - :func:`run<nvflare.app_common.workflows.model_controller.ModelController.run>`
+   * - send_model_and_wait
+     - Send a task with data to targets (blocking) and wait for results..
+     - :func:`send_model_and_wait<nvflare.app_common.workflows.model_controller.ModelController.send_model_and_wait>`
+   * - send_model
+     - Send a task with data to targets (non-blocking) with callback.
+     - :func:`send_model<nvflare.app_common.workflows.model_controller.ModelController.send_model>`
+   * - sample_clients
+     - Returns a list of num_clients clients.
+     - :func:`sample_clients<nvflare.app_common.workflows.model_controller.ModelController.sample_clients>`
+   * - save_model
+     - Save model with persistor.
+     - :func:`save_model<nvflare.app_common.workflows.model_controller.ModelController.save_model>`
+   * - load_model
+     - Load model from persistor.
+     - :func:`load_model<nvflare.app_common.workflows.model_controller.ModelController.load_model>`
+
+
+Communication
+=============
+
+send_model_and_wait
+-------------------
+:func:`send_model_and_wait<nvflare.app_common.workflows.model_controller.ModelController.send_model_and_wait>` is the core communication function which enables users to send tasks to targets, and wait for responses.
+
+The ``data`` is an :ref:`fl_model` object, and the ``task_name`` is the task for the target executors to execute (Client API executors by default support "train", "validate", and "submit_model", however executors can be written for any arbitrary task name).
+
+``targets`` can be chosen from client names obtained with ``sample_clients()``.
+
+Returns the :ref:`fl_model` responses from the target clients once the task is completed (``min_responses`` have been received, or ``timeout`` time has passed).
+
+send_model
```

Comment:
Right this is a good point, @YuanTingHsieh and I were also discussing about these options too. 

For (1), rather than introducing a separate FLTask object, maybe we can instead rename `send_model` to either `send` or `send_task` to be more clear that we are sending tasks? (as we are always sending tasks, not just in KM and CSE) 

For (2) I agree it is good to emphasize that the FLModel `data` is specific to the task being sent and may or may not be an actual ML model, but is rather an optional data structure paired with the task.

Reply:
I like `send_task`.

@holgerroth @ZiyueXu77 do you have any preferences?

Reply:
Since we need FLModel defined to be "sent", I think send_model makes sense for current pipeline. I would prefer @chesterxgchen's option 2 - explain that the action of send_model is in fact dispatching a task with the "core content" as the model. Otherwise if renaming to "send_task", we may need to further define FLTask(model= , ...) to pair with it.